File: packages/zynth-crypto/android/ZynthCrypto/src/main/java/dev/zynth/crypto/ZynthCrypto.kt
package dev.zynth.crypto

import android.app.Activity
import com.zynth.kit.runtime.ZynthRuntime

object ZynthCrypto {
  private var moduleInstance: ZynthCryptoModule? = null

  @JvmStatic
  fun initialize(activity: Activity, runtime: ZynthRuntime) {
    if (moduleInstance != null) {
      return
    }

    val module = ZynthCryptoModule()
    runtime.installModules(listOf(module))
    moduleInstance = module
  }

  @JvmStatic
  fun cleanup() {
    moduleInstance = null
  }
}

File: packages/zynth-crypto/ios/ZynthCrypto.swift
import Foundation
import ZynthKit

@objc(ZynthCrypto)
public final class ZynthCrypto: NSObject {
  private static var module: ZynthCryptoModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard module == nil else {
      return
    }

    let instance = ZynthCryptoModule()
    runtime.installModules([instance])
    module = instance
  }

  @objc public static func cleanup() {
    module = nil
  }
}
